app/main.py

```diff
@@ -1,2 +1,50 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+    def __init__(self) -> None:
+        self.length = 0
+        self.capacity = 8
+        self.hash_table = [None] * self.capacity
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = hash(key) % self.capacity
+        if self.length >= round(self.capacity * 2 / 3) - 1:
```

Comment:
You should make resize at the end of `__setitem__` method to make it more clear
Then you will call self.resize if length >= threshold, without additional `- 1`